### PR TITLE
Add Performance APIs

### PR DESCRIPTION
**Pull request by Elchi3**
Web features for the Performance APIs, see https://developer.mozilla.org/en-US/docs/Web/API/Performance_API

Adds a new "Performance" group consisting of these features:
- Performance (main feature, uses compute_from)
- Back/forward cache blocking reasons
- Element timing
- Event timing
- Largest contentful paint (LCP)
- Layout instability
- Long tasks
- Memory measurement
- Navigation timing
- Page visibility state
- Paint timing
- Resource timing (initial support)
- Resource size
- Server timing

### CODE REVIEW

File: features/page-visibility.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: Page visibility state
+description: The `"visibility-state"` performance entry and the `VisibilityStateEntry` interface provide timings of page visibility state changes.
```
Comment (autonome) [inline features/page-visibility.yml:2]: The extra quotes don't seem necessary.

File: features/page-visibility.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: Page visibility state
```
Comment (autonome) [inline features/page-visibility.yml:1]: might want to preclude a future battle for that filename when we get to `api.Document.visibilityState`, known as the Page Visibility API.

**Review by autonome — COMMENTED** [review]
First pass review, but it's a big PR, so should get another review as well.

- 👍🏼 on the group rename
- 👍🏼 on removing `performance` from the feature names
- 👍🏼 on overall feature breakdown/composition
- The descriptions seemed generally fine but IIRC we've tended away from using `interface` and instead making active descriptions of the implementation, should get a 2nd opinion here

File: features/measure-memory.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+name: Memory measurement
+description: The `measureUserAgentSpecificMemory()` method is used to estimate the memory usage of a web application including all its iframes and workers.
```
Comment (ddbeck) [inline features/measure-memory.yml:2]:
```suggestion
description: The `measureUserAgentSpecificMemory()` method estimates the memory usage of a web application including all its iframes and workers.
```

File: features/longtasks.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
-draft_date: 2024-09-09
-name: Long Tasks API
-description: TODO
+name: Long tasks
+description: The `PerformanceLongTaskTiming` interface provides information about tasks that occupy the UI thread for 50 milliseconds or more.
```
Comment (ddbeck) [inline features/longtasks.yml:2]: For symmetry with the next few features:

```suggestion
description: The `longtask` performance event type and the `PerformanceLongTaskTiming` API provides information about tasks that occupy the UI thread for 50 milliseconds or more, a metric for input latency, delayed animations, and other sources of "jank."```

File: features/layout-instability.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
-draft_date: 2024-09-09
-name: Layout Instability
-description: TODO
+name: Layout instability
+description: The `LayoutShift` interface provides a way to measure and report on the layout stability of web pages based on movements of the elements on the page.
```
Comment (ddbeck) [inline features/layout-instability.yml:2]: Consistency with other features like this:

```suggestion
description: The `layout-shift` performance entry and `LayoutShift` API measures the layout stability of web pages based on movements of the elements on the page.
```

File: features/largest-contentful-paint.yml [hunk 6]
```diff
@@ -1,7 +1,7 @@
-draft_date: 2024-09-09
-name: Largest Contentful Paint
-description: TODO
+name: Largest contentful paint (LCP)
+description: The `LargestContentfulPaint` interface provides timing information about the largest image or text paint before user input on a web page. The key moment this API provides is known as the LCP metric.
```
Comment (ddbeck) [inline features/largest-contentful-paint.yml:2]: For symmetry with other features like this:

```suggestion
description: The `largest-contentful-paint` performance entry and the `LargestContentfulPaint` API measures  the time it takes for the largest image or text to appear. Largest contentful paint (LCP) is a common metric for perceived loading times.
```

File: features/event-timing.yml [hunk 7]
```diff
@@ -1,7 +1,9 @@
-draft_date: 2024-09-09
-name: Event Timing API
+name: Event timing
 description: TODO
```
Comment (ddbeck) [inline features/event-timing.yml:2]: For consistency with other features like this:

```suggestion
description: The `event` and `first-input` performance entries and the `PerformanceEventTiming` API measures the latency of user input events, such as mouse clicks or keypresses, and the latency of the first user input. First Input Delay (FID) is a common metric for perceived responsiveness.
```

File: features/element-timing.yml [hunk 8]
```diff
@@ -1,7 +1,7 @@
-draft_date: 2024-09-09
-name: Element Timing API
-description: TODO
+name: Element timing
+description: The `PerformanceElementTiming` interface contains render timing information for image and text node elements the developer annotated with an `elementtiming` attribute for observation.
 spec: https://wicg.github.io/element-timing/
+group: performance
 compat_features:
```
Comment (ddbeck) [inline features/element-timing.yml:5]: There should probably be something like `html.global_attributes.elementtiming` in here, right?

File: features/element-timing.yml [hunk 9]
```diff
@@ -1,7 +1,7 @@
-draft_date: 2024-09-09
-name: Element Timing API
-description: TODO
+name: Element timing
+description: The `PerformanceElementTiming` interface contains render timing information for image and text node elements the developer annotated with an `elementtiming` attribute for observation.
```
Comment (ddbeck) [inline features/element-timing.yml:2]:
```suggestion
description: The `element` performance entry and the `PerformanceElementTiming` API measures the time to render elements that have the  `elementtiming` attribute.
```

File: features/bfcache-blocking-reasons.yml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+name: Back/forward cache blocking reasons
+description: The `PerformanceNavigationTiming.notRestoredReasons` property reports information on why the current document was blocked from using the back/forward cache (bfcache) on navigation.
```
Comment (ddbeck) [inline features/bfcache-blocking-reasons.yml:2]:
```suggestion
description: The `notRestoredReasons` property of a `PerformanceNavigationTiming` object explains why the current document was blocked from using the back/forward cache (also known as the bfcache) on navigation.
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
A bunch of ideas for you!

File: features/performance.yml [hunk 11]
```diff
@@ -0,0 +1,49 @@
+name: Performance
+description: The global `performance` property and the `Performance` interface provide access to performance-related information for the current execution context.
+spec:
+  - https://w3c.github.io/hr-time/
+  - https://w3c.github.io/performance-timeline/
+  - https://w3c.github.io/resource-timing/
+  - https://w3c.github.io/user-timing/#extensions-performance-interface
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply features/performance.yml:7]: Yes, I think both would map to this feature. 
- `user-timing` is basically `PerformanceMark` and `PerformanceMeasure`
- `high-resolution-time` is basically `performance.now`

File: features/element-timing.yml (hunk 8 above)
Comment (Elchi3) [reply features/element-timing.yml:5]: hm, there isn't such a key in BCD. The docs have no compat table either. https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/elementtiming. Not sure if it is "global" either:

> This attribute may be applied to ``, `<image>` elements inside an `<svg>`, poster images of `<video>` elements, elements which have a background-image, and elements containing text nodes, such as a `<p>`.

File: features/performance.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (Elchi3) [reply features/performance.yml:7]: Added in https://github.com/web-platform-dx/web-features/pull/1934/commits/a8d3bf654e2196826a293631b8533d99711dc837

**Review by ddbeck — APPROVED** [review]
This is ready, apart from some messes I made.

I caused the conflicts by merging a change to the specs drafts script. Merging `main`, running `update-drafts` on the affected specs, and refreshing the dist should fix it up nicely.

I also suggested some fixes to typos that I introduced.

**Comment by Elchi3** [comment]
Thanks Daniel, I think I also caused a bit of a mess by manually updating drafts plus the latest BCD release actually contained new Performance API keys plus the new draft collection now also contains sub features without a spec_url.

In theory it should be all fixed now and we can be sure that all keys have been mapped!

File: features/element-timing.yml (hunk 8 above)
Comment (ddbeck) [reply features/element-timing.yml:5]: It turns out there's no actual specification for this (though its reflection is via `Element`). I filed https://github.com/mdn/browser-compat-data/issues/24788

**Review by ddbeck — APPROVED** [review]

**Pull request merged**